init.sls

```diff
@@ -72,20 +72,31 @@ chown_pgdata:
 
 # Deploy users. Sort them, so order is not changing between salt runs
 {% for config in pillar['postgresql']['users']|default({}) %}
-createuser-{{ loop.index }}:
+createuser-{{ config['username'] }}:
   cmd.run:
     - name: psql -t -c "CREATE ROLE {{ config['username'] }} {{ config['options']|default('NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN') }};"
     - unless: psql -t -c "SELECT 1 FROM pg_roles WHERE rolname='{{ config['username'] }}'" |grep -q 1
     - runas: postgres
+{% endfor %}
 
-# The "replication" keyword is not a real database, but just used for permissions in pg_hba.conf
+{% for config in pillar['postgresql']['databases']|default({}) %}
 {% if config['database'] != "replication" %}
-createdb-{{ loop.index }}:
+createdb-{{ config['database'] }}:
```

Comment:
I like the idea of having additional databases. Unfortunately, this would break existing configurations, were databases where created using the following pillar fields:

```yaml
users: 
  - username: test
    database: test
```

Is there a simple way to provide both syntaxes? Otherwise, a migration guide needs to be written (in README or CHANGELOG).

Another alternative would be to allow multiple databases per user, e.g. like this:

```yaml
users:
  - username: test
    databases: [bla, blubb]
```

The `database: ''` and `databases: []` pillars could even be merged automatically (not sure if this is necessary though, I suppose just supporting `databases` as an array would be sufficient).